packages/mdl-drawer/README.md

```diff
@@ -0,0 +1,284 @@
+
+#### Using an existing element.
+
+If you do not care about retaining the component instance for the temporary drawer, simply call `attachTo()`
+and pass it a DOM element.  
+
+```javascript
+mdl.TemporaryDrawer.attachTo(document.querySelector('.mdl-temporary-drawer'));
+```
+
+#### Manual Instantiation
+
+Temporary drawers can easily be initialized using their default constructors as well, similar to `attachTo`.
+
+```javascript
+import MDLTemporaryDrawer from 'mdl-temporary-drawer';
+
+const drawer = new MDLTemporaryDrawer(document.querySelector('.mdl-temporary-drawer'));
+```
+
+### Using the Foundation Class
+
+MDL Temporary Drawer ships with an `MDLTemporaryDrawerFoundation` class that external frameworks and libraries can
+use to integrate the component. As with all foundation classes, an adapter object must be provided.
+The adapter for temporary drawers must provide the following functions, with correct signatures:
+
+| Method Signature | Description |
+| --- | --- |
+| `addClass(className: string) => void` | Adds a class to the root element. |
+| `removeClass(className: string) => void` | Removes a class from the root element. |
+| `hasClass(className: string) => boolean` | Returns boolean indicating whether element has a given class. |
+| `hasNecessaryDOM() => boolean` | Returns boolean indicating whether the necessary DOM is present (namely, the drawer container). |
+| `registerComponentInteractionHandler(evt: string, handler: EventListener) => void` | Adds an event listener to the root element, for the specified event name. |
+| `deregisterComponentInteractionHandler(evt: string, handler: EventListener) => void` | Removes an event listener from the root element, for the specified event name. |
+| `registerDrawerInteractionHandler(evt: string, handler: EventListener) => void` | Adds an event listener to the drawer container sub-element, for the specified event name. |
+| `deregisterDrawerInteractionHandler(evt: string, handler: EventListener) => void` | Removes an event listener from drawer container sub-element, for the specified event name. |
+| `registerTransitionEndHandler(handler: EventListener) => void` | Registers an event handler to be called when a `transitionend` event is triggered on the drawer container sub-element element. |
+| `deregisterTransitionEndHandler(handler: EventListener) => void` | Deregisters an event handler from a `transitionend` event listener. This will only be called with handlers that have previously been passed to `registerTransitionEndHandler` calls. |
+| `getDrawerWidth() => number` | Returns the current drawer width, in pixels. |
+| `updateDrawerPosition(value: number) => void` | Sets the current position for the drawer, in pixels from the border. |
+| `updateComponentBackground(value: string) => void` | Sets a CSS custom property, for controlling the current background opacity when manually dragging the drawer. |
```

Comment:
Should we specify what this custom property is? Or, the ripple uses the method name `updateCssVariable` for this. Should we try and keep it consistent?